### PR TITLE
Test templates

### PR DESCRIPTION
**Pull request by adborden**
**Not for merge**

Just want to get some feedback on this testing approach. It uses the `ApiClient`'s filesystem feature. By setting `API_BASE` to a filepath, files will be read off the filesystem rather than via an HTTP API. For the one partial, 10 API endpoints are called, which is a lot of data to mock. I think the filesystem is be easier to deal with this fixture data, rather than stub individual endpoints/methods in the test code.

I think this is a reasonable approach for now, and we can revisit once we get an idea of what kinds of templates we'll be overriding and what kinds of things we need to test within `fec-eregs`. Some other thoughts:
- I'm thinking a lot of test cases will come out of specific cases we see "in the wild", so creating fixtures based on real examples works well with this approach.
- Not all the layers are necessary to render, so I emptied the ones that weren't essential. Some of these files would've been thousands of lines otherwise. It might make sense to modify the `ApiClient` to be more flexible to accept missing files, so we don't even need the empty `{}` files to exist.
- Maybe pull the filesystem code into it's own TestApiClient, and expose it via a module from regulations-site. I'd love to see regulations-site provide some test utilities so that other eregs implementations have a toolbox to test template modifications.
- I could add a script to generate the fixture files by fetching data from the api given an endpoint and dumping it to a directory.

### CODE REVIEW

File: requirements_test.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mock==1.3.0
+beautifulsoup4==4.3.2
```
Comment (adborden) [inline requirements_test.txt]: These aren't needed, I'll remove them.

**Comment by cmc333333** [comment]
This feels very heavy for a non-integration test. I don't think you really need to test that data is being loaded from the API correctly, that Django is compiling a view accurately, the HTMLBuilder is applying, etc. etc. -- that's already tested elsewhere.

You're really just testing the template here, so would it make sense to do something simpler, a la https://github.com/18F/regulations-site/blob/master/regulations/tests/views_partial_definitions_tests.py ? Speaking of, it'd be even easier if this made it's way into -site.

**Comment by adborden** [comment]
Thanks! Yeah, not having a great grasp (yet) of the internals of the eregs modules makes it tough to make those kinds of calls, so I was erring on the side of caution to test all the things. I'll go with the narrow approach for now.

**Comment by adborden** [comment]
@cmc333333 I didn't understand this last bit:

> Speaking of, it'd be even easier if this made it's way into -site.

These tests are for a [modified definition template](https://github.com/18F/fec-eregs/blob/master/fec_eregs/templates/regulations/partial-definition.html) which shows a single paragraph when the definition is ambiguous. Are you saying that behavior is desired in -site?

**Comment by cmc333333** [comment]
@adborden exactly. I can't think of a reason that this would be FEC-specific functionality.

**Pull request closed without merging**